sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -240,19 +242,41 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
 
   /**
    * Update DisbursementSchedule with documentNumber
-   * @param documentNumber document Number
+   * The update to Application and Disbursement schedule happens in single transaction.
+   * @param disbursementScheduleId document Number
    * @returns the result of update.
    */
-  async updateDisbursementScheduleDocumentNumber(
+  async updateDisbursementAndApplicationCOEApproval(
+    disbursementScheduleId: number,
+    userId: number,
     applicationId: number,
-  ): Promise<UpdateResult> {
+    applicationStatus: ApplicationStatus,
+  ): Promise<void> {
     const documentNumber = await this.getNextDocumentNumber();
-    return this.repo
-      .createQueryBuilder()
-      .update(DisbursementSchedule)
-      .set({ documentNumber: documentNumber })
-      .where("application.id = :applicationId", { applicationId })
-      .execute();
+    this.connection.transaction(async (transactionalEntityManager) => {
```

Comment:
I believe that we need to await here or return the Promise to the consumer await on it.